synapse/api/auth.py

```diff
@@ -79,7 +79,7 @@ def __init__(self, hs):
 
         self._auth_blocking = AuthBlocking(self.hs)
 
-        self._account_validity = hs.config.account_validity
+        self._account_validity_enabled = hs.config.account_validity_enabled
```

Comment:
I won't demand you change this, since it's a backport, but we really ought to be using the fully-specified names for the config options rather than adding more places we fall back on the grandfathered magic:

```suggestion
        self._account_validity_enabled = hs.config.account_validity.account_validity_enabled
```

The idea is that we stop using the magic grandfathering, then remove the magic, then we can give these things less cumbersome names.

Reply:
Oh, right! I think this was before I was really aware of it. This has a slight performance improvement as well, correct?

Reply:
correct.

Reply:
Done in bd15afef24!